includes/class-sensei-usage-tracking-data.php

```diff
@@ -331,6 +338,71 @@ private static function get_course_featured_count() {
 		return $query->found_posts;
 	}
 
+	/**
+	 * Gets the total number of non-admin learners enrolled in at least one published course.
+	 *
+	 * @since 1.12.2
+	 *
+	 * @return int Number of course enrolments.
+	 **/
+	private static function get_course_enrolments() {
+		global $wpdb;
+
+		return $wpdb->get_var(
+			"SELECT COUNT(DISTINCT c.user_id)
+			FROM {$wpdb->comments} c
+			INNER JOIN {$wpdb->usermeta} um ON c.user_id = um.user_id
+			INNER JOIN {$wpdb->posts} p ON p.ID = c.comment_post_ID
+			WHERE comment_type = 'sensei_course_status'
+				AND meta_key = '{$wpdb->prefix}capabilities' AND meta_value NOT LIKE '%administrator%'
+				AND post_status = 'publish' AND c.user_id <> 0"
```

Comment:
The fact that we're explicitly ignoring a `user_id` of `0` here means that we're not counting anonymous users (i.e. users who take courses that do not require a login). Are we ok with that?

Reply:
I don't believe that's what a user with an ID of 0 represents. If you're not logged in, you shouldn't be allowed to take a course at all. The "Start Taking This Course" button just isn't available.

I'd seen these "phantom" entries when testing locally (i.e. where the Leaner name is empty), and they actually look like corrupt data to me, so I'd excluded them for that reason. My hunch is these entries appear when you import course data from another site with a different list of users.

Incidentally, the learners count also has this check in its query.

Reply:
Oh...For some reason, I thought that there was a way to take a course without logging in. But I just tested it and your right. I stand corrected! 🙂